lib/vanagon/platform/rpm.rb

```diff
@@ -40,6 +49,13 @@ def output_dir(target_repo = "products")
         super
       end
 
+      # Method to derive the directory for source artifacts
+      #
+      # @param target_repo [String] repo the source artifacts are targeting
+      def source_output_dir(target_repo = "products")
+        @source_output_dir ||= File.join(@os_name, @os_version, target_repo, 'SRPMS')
```

Comment:
most of those don't have a target repo set and for RPMs that defaults to target repo of 'products' (so, like http://builds.puppetlabs.lan/puppet-enterprise-vanagon/pe-activemq-2016.4.5.14.3-3/artifacts/el/7/products/x86_64/) and debs defaults to no target repo (like http://builds.puppetlabs.lan/puppet-enterprise-vanagon/pe-activemq-2016.4.5.14.3-3/artifacts/deb/precise/)